### PR TITLE
Publish the logic used by the CLI as a library

### PR DESCRIPTION
**Pull request by bmbferreira**
* Refactor to separate the code to be released as a library from the CLI.
* Sbt changes to publish the two components (CLI and Core) separately.

### CODE REVIEW

**Comment by rtfpessoa** [comment]
Is it still WIP?

**Comment by bmbferreira** [comment]
no

File: .circleci/config.yml [hunk 1]
```diff
@@ -187,7 +188,32 @@ jobs:
...
+          name: Setup AWS Credentials
+          command: ./scripts/setup-aws-credentials.sh
+      - run:
+          name: Retrieve GPG Keys
+          command: |
+            mkdir -p ~/.gnupg
+            aws s3 sync --profile shared-services --include '*.gpg' s3://$AWS_CREDENTIALS_BUCKET/gnupg/ ~/.gnupg
+      - run:
+          name: Publish to Sonatype
+          command: ./scripts/publish.sh 0.1.$CIRCLE_BUILD_NUM
```
Comment (rtfpessoa) [inline .circleci/config.yml]: This should be the same as in the CLI `export PUBLISH_VERSION="0.1.0-alpha3.$CIRCLE_BUILD_NUM"`.
Maybe move this to the common stuff in the top.

File: build.sbt [hunk 2]
```diff
@@ -32,10 +38,75 @@ lazy val codacyAnalysisCli = project
...
+      sys.env.getOrElse("SONATYPE_PASSWORD", "password")),
+    publishMavenStyle := true,
+    publishArtifact.in(Test) := false,
+    publishArtifact.in(makePom.in(Docker)) := false,
+    publish.in(Docker) := {},
+    publishLocal.in(Docker) := {},
+    pomIncludeRepository := { _ =>
+      false
+    },
+    publishTo := {
```
Comment (rtfpessoa) [inline build.sbt]: Can you replace with https://github.com/xerial/sbt-sonatype#buildsbt

File: .circleci/config.yml [hunk 3]
```diff
@@ -6,6 +6,9 @@ references:
   host_environment_defaults: &host_environment_defaults
     # Customize the JVM maximum heap limit
     JAVA_OPTS: -Xmx3200m
+    # Publish version number for the CLI docker and library
+    PUBLISH_VERSION: "0.1.0-alpha3.$CIRCLE_BUILD_NUM"
```
Comment (rtfpessoa) [inline .circleci/config.yml]: This does not work. Env variables cannot have other variables

**Pull request merged**